crates/rethnet_evm/benches/state_debug.rs

```diff
@@ -0,0 +1,330 @@
+    bench_sync_state_method(
+        c,
+        "StateDebug::account_storage_root(), account exists",
+        prep_no_op,
+        |state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            let result = state.account_storage_root(&address);
+            debug_assert!(result.is_ok());
+            debug_assert!(result.unwrap().is_some());
+        },
+    );
+}
+
+fn bench_insert_account_already_exists(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::insert_account(), account already exists",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            let result = state.insert_account(address, AccountInfo::default());
+            debug_assert!(result.is_ok())
+        },
+    );
+}
+
+fn bench_insert_account_doesnt_exist(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::insert_account(), account doesn't yet exist",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts + 1);
+            debug_assert!(state.basic(address).unwrap().is_none());
+            let result = state.insert_account(address, AccountInfo::default());
+            debug_assert!(result.is_ok());
+        },
+    );
+}
+
+fn bench_modify_account_doesnt_exist(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::modify_account(), account doesn't exist",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts + 1);
+            debug_assert!(state.basic(address).unwrap().is_none());
+            let result = state.modify_account(
+                address,
+                AccountModifierFn::new(Box::new(|_balance, &mut _nonce, _code| {})),
+                &|| Ok(AccountInfo::default()),
+            );
+            debug_assert!(result.is_ok());
+        },
+    );
+}
+
+fn bench_modify_account_exists_with_code_no_change(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::modify_account(), account already exists, with code, no code change",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            // TODO: figure out why the following assert is failing
+            //debug_assert!(state.basic(address).unwrap().unwrap().code.is_some());
+            let result = state.modify_account(
+                address,
+                AccountModifierFn::new(Box::new(|_balance, &mut _nonce, _code| {})),
```

Comment:
The modify functions are not actually changing anything, which might lead to a no-op. I recommend at least incrementing the nonce and doing a balance tweak.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3910/commits/1ef6815ef0e13fba9048535a4dbb3ebc271e7fb9